File: internal/sast/fixtures/python/log_negative.py
# internal/sast/fixtures/python/log_negative.py
import logging
from flask import Flask, request

app = Flask(__name__)
log = logging.getLogger(__name__)

@app.route("/login")
def login():
    user = (request.args.get("user", "") or "").replace("\n", "").replace("\r", "")[:64]
    log.info("Login attempt: %s", user)
    return "ok"

File: internal/sast/fixtures/javascript/log_negative.js
// internal/sast/fixtures/javascript/log_negative.js
const express = require("express");
const app = express();

app.get("/login", (req, res) => {
  const safe = String(req.query.user || "").replace(/[\r\n]/g, "").slice(0, 64);
  console.log("Login attempt: %s", safe);
  res.send("ok");
});
